lib/models/schema.js

```diff
@@ -17,6 +17,11 @@ const MixinSpecificationExtensions = require('../mixins/specification-extensions
  * @returns {Schema}
  */
 class Schema extends Base {
+  constructor (json) {
+    super(json === false ? {} : json);
+    this._json = json;
```

Comment:
Would it not be possible to change the `base` check to check for undefined instead of booleans as well?🤔 Or would that create problems? 🤔 

Setting `_json` twice seems like the wrong approach 😅

Reply:
The base ctor is has:

```js
    if (!json) throw new ParserError(`Invalid JSON to instantiate the ${this.constructor.name} object.`);
    this._json = json;
```

I don't want to change it, because handling `false` schemas in base class isn't a good idea. At the moment it's only needed for `Schema` so I pass empty object as workaround

> Setting _json twice seems like the wrong approach 😅

Yep, but it's still "fine" if we make this assignment in child ctor, not in other place :)

Reply:
Also checking undefined is not good idea - JSON Schema hasn't `undefined` type/value

Reply:
To my understanding, the [base class constructor](https://github.com/asyncapi/parser-js/blob/269206c564cb1271a56771676b8f71601d80ba6c/lib/models/base.js#L11) check, in theory, should only make sure that the received value is not undefined/null 🤔 

Therefore you could change that referenced line from a lazy check to strict:

```js
    if (json === undefined || json === null) throw new ParserError(`Invalid JSON to instantiate the ${this.constructor.name} object.`);
```

That way you can remove `this._json = json;` in this constructor.

Reply:
I might be wrong though 😄

Reply:
Perhaps I'm sitting in work too long today 😅 Thanks! Changed.

Reply:
haha 😆 That or I have, lets see if others find a problem with this approach 😆

Reply:
but even if you put `""` or `0` what it's a value in spec? Can we pass something like this in spec itself? Only `false` has some "value", but only related to Schema object.

Reply:
Not that I know of no, hence my suggestion. 

The only dilemma I can see could be if the JSON Schema for the specification allowed such values where objects are expected, then it would no longer throw a parser error here. I just don't see the use-case 🤷

Reply:
I also 😅